pkg/ocm/helpers.go

```diff
@@ -734,7 +734,7 @@ func ValidateHostedClusterSubnets(awsClient aws.Client, isPrivate bool, subnetID
 	}
 
 	privateSubnetCount := len(privateSubnets)
-	publicSubnetsCount := len(vpcSubnets) - privateSubnetCount
+	publicSubnetsCount := len(subnets) - privateSubnetCount
```

Comment:
@mikeshng Can we back this up with a unit test in `helpers_test.go`? 

I agree with your changes. I would like us to spend time now to write a test that ensures that we don't inadvertently break this again in the future as our lack of coverage is one of the major reasons why we have this type of regression.

If we add coverage as we go, it makes larger refactors easier in the future as we have a contract that tells us if we're breaking things.

Let me know if you want any help with the test. `Client` interface for AWS is mocked via `gomock` and available in `pkg/aws/mock_client.go`

cheers,

Rob

Reply:
Hi @robpblake thank you for the review and feedback. I've done as you suggested. Could you PTAL again? Thanks.